src/widgets/modals/SearchForm.js

```diff
@@ -97,61 +127,54 @@ export const SearchFormComponent = ({
     return () => null;
   }, [isPatient, isPrescriber]);
 
-  const lookupPatient = useCallback(params => {
-    setIsQuerying(true);
-    queryPatientApi(params).then(({ error: patientError, data: patientData }) => {
-      if (patientError) {
-        setIsError(true);
-        setError(patientError);
-      }
-      setData(patientData);
-      setIsQuerying(false);
-    });
-  }, []);
+  const FormView = useCallback(
+    () => (
+      <FormControl
+        inputConfig={formConfig}
+        onSave={lookupRecords}
+        showCancelButton={false}
+        saveButtonText={generalStrings.search}
+      />
+    ),
+    [formConfig, lookupRecords]
+  );
 
-  const lookupPrescriber = useCallback(params => {
-    setIsQuerying(true);
-    queryPrescriberApi(params).then(({ error: prescriberError, data: prescriberData }) => {
-      if (prescriberError) {
-        setIsError(true);
-        setError(prescriberError);
-      }
-      setData(prescriberData);
-      setIsQuerying(false);
-    });
-  }, []);
+  const ListView = useCallback(
+    () =>
+      isQueryUrl ? (
+        <QueryHandler queryUrl={queryUrl} onData={onData} onError={onError} />
+      ) : (
+        <FlatList data={data} keyExtractor={record => record.id} renderItem={renderRecord} />
```

Comment:
Hmm, because of the way the hook works, the `QueryHandler` basically needs to be mounted when the query is being performed, and unmounted when it is completed.

Not sure if it's great tbh, maybe needs a revist, but works for now!